tests/unit_tests/dft/source/workspace_external_tests.cpp

```diff
@@ -0,0 +1,358 @@
+    using namespace oneapi::mkl::dft;
+    using scalar_t = std::conditional_t<prec == precision::DOUBLE, double, float>;
+    using forward_t = std::conditional_t<dom == domain::COMPLEX, std::complex<scalar_t>, scalar_t>;
+    using backward_t = std::complex<scalar_t>;
+
+    sycl::queue sycl_queue(*dev);
+    if (prec == precision::DOUBLE && !sycl_queue.get_device().has(sycl::aspect::fp64)) {
+        return test_skipped;
+    }
+    descriptor<prec, dom> desc(static_cast<std::int64_t>(dftSize));
+
+    desc.set_value(config_param::WORKSPACE_PLACEMENT, config_value::WORKSPACE_EXTERNAL);
+    desc.set_value(config_param::PLACEMENT, config_value::NOT_INPLACE);
+    commit_descriptor(desc, sycl_queue);
+    std::int64_t workspace_bytes = -1;
+    desc.get_value(config_param::WORKSPACE_EXTERNAL_BYTES, &workspace_bytes);
+    if (workspace_bytes < 0) {
+        return ::testing::Test::HasFailure();
+    }
+    scalar_t* workspace = sycl::malloc_device<scalar_t>(
+        static_cast<std::size_t>(workspace_bytes) / sizeof(scalar_t), sycl_queue);
+    desc.set_workspace(workspace);
+    // Generate data
+    std::vector<forward_t> hostFwd(static_cast<std::size_t>(dftSize));
+    std::size_t bwdSize = dom == domain::COMPLEX ? dftSize : dftSize / 2 + 1;
+    std::vector<backward_t> hostBwd(bwdSize);
+    rand_vector(hostFwd, dftSize);
+
+    // Allocate enough memory that we don't have to worry about the domain.
+    forward_t* deviceFwd = sycl::malloc_device<forward_t>(dftSize, sycl_queue);
+    backward_t* deviceBwd = sycl::malloc_device<backward_t>(bwdSize, sycl_queue);
+    sycl_queue.copy(hostFwd.data(), deviceFwd, dftSize);
+    sycl_queue.wait_and_throw();
+
+    compute_forward<decltype(desc), forward_t, backward_t>(desc, deviceFwd, deviceBwd);
+    sycl_queue.wait_and_throw();
+
+    sycl_queue.copy(deviceBwd, hostBwd.data(), bwdSize);
+    sycl_queue.wait_and_throw();
+
+    // To see external workspaces, larger sizes of DFT may be needed. Using the reference DFT with larger sizes is slow,
+    // so use Parseval's theorum as a sanity check instead.
+    bool sanityCheckPasses = parseval_check(dftSize, hostFwd.data(), hostBwd.data());
+
+    if (sanityCheckPasses) {
+        sycl_queue.copy(hostFwd.data(), deviceFwd, dftSize);
+        sycl_queue.wait_and_throw();
+        compute_backward<decltype(desc), backward_t, forward_t>(desc, deviceBwd, deviceFwd);
+        sycl_queue.wait_and_throw();
+        sycl_queue.copy(deviceFwd, hostFwd.data(), dftSize);
+        sycl_queue.wait_and_throw();
+        forward_t rescale =
+            static_cast<forward_t>(1) / static_cast<forward_t>(static_cast<scalar_t>(dftSize));
+        sanityCheckPasses = parseval_check(dftSize, hostFwd.data(), hostBwd.data(), rescale);
+    }
+
+    sycl::free(deviceFwd, sycl_queue);
+    sycl::free(deviceBwd, sycl_queue);
+    sycl::free(workspace, sycl_queue);
+    return sanityCheckPasses ? !::testing::Test::HasFailure() : ::testing::Test::HasFailure();
+    ;
```

Comment:
```suggestion
```

Reply:
Done in [1da13ee](https://github.com/oneapi-src/oneMKL/pull/411/commits/1da13eef0fbb663fcadc1abbcc15ad930f977589).